pkg/reconciler/pipelinerun/pipelinerun_test.go

```diff
@@ -1743,16 +1739,44 @@ spec:
     name: test-pipeline
   serviceAccountName: test-sa
 status:
+  conditions:
+  - message: running...
+    reason: Running
+    status: Unknown
+    type: Succeeded
   startTime: "2021-12-31T00:00:00Z"
+  runs:
+    test-pipeline-run-custom-task-hello-world-1:
+      pipelineTaskName: hello-world-1
+      status:
+        conditions:
+        - status: Unknown
+          type: Succeeded
 `)}
 			prs[0].Spec.Timeout = tc.timeout
 			prs[0].Spec.Timeouts = tc.timeouts
 
+			runs := []*v1alpha1.Run{mustParseRunWithObjectMeta(t,
```

Comment:
As far as I know this is the only timeout test where the existing runs/taskruns have already been created. (The other ones have the runs/taskruns not created yet.) I think we should test the cases where runs/taskruns have already been created, and are canceled after timeouts.tasks and timeouts.finally.

Reply:
Good point. I'll add that.